app/services/query-manager.js

```diff
@@ -150,6 +161,51 @@ export default Service.extend({
     });
   },
 
+  replaceWindow(query, emitType, emitEvery, includeType) {
+    return query.get('window').then(window => {
+      let emit = EmberObject.create({
+        type: window.get('emit.type'),
+        every: window.get('emit.every')
+      });
+      this.setIfNotEmpty(emit, 'type', emitType);
+      this.setIfNotEmpty(emit, 'every', emitEvery);
+      let include = EmberObject.create({
+        type: window.get('include.type')
+      });
+      this.setIfNotEmpty(include, 'type', includeType);
+      window.set('emit', emit);
+      window.set('include', include);
+      return window.save();
+    });
+  },
+
+  deleteWindow(query) {
+    return query.get('window').then(window => {
+      if (isEmpty(window)) {
+        return resolve();
+      }
+      query.set('window', null);
+      return query.save().then(() => {
+        return window.destroyRecord();
+      });
+    });
+  },
+
+  addWindow(query) {
+    return query.get('window').then(window => {
+      if (!isEmpty(window)) {
```

Comment:
Do we need this check? Can this be called with an existing window? It feels like the caller should be the one doing the checking if necessary